clang-tools-extra/clangd/refactor/Rename.cpp

```diff
@@ -778,6 +747,45 @@ void findNearMiss(
 
 } // namespace
 
+llvm::Expected<FileEdits>
+editsForLocations(const llvm::StringMap<std::vector<Range>> &Ranges,
+                  const SymbolName &OldName, const SymbolName &NewName,
+                  llvm::vfs::FileSystem &FS, const LangOptions &LangOpts) {
+  FileEdits Results;
+  for (auto &FileAndOccurrences : Ranges) {
+    llvm::StringRef FilePath = FileAndOccurrences.first();
+
+    auto ExpBuffer = FS.getBufferForFile(FilePath);
+    if (!ExpBuffer) {
+      elog("Fail to read file content: Fail to open file {0}: {1}", FilePath,
+           ExpBuffer.getError().message());
+      continue;
+    }
+
+    auto AffectedFileCode = (*ExpBuffer)->getBuffer();
+    syntax::UnexpandedTokenBuffer Tokens(AffectedFileCode, LangOpts);
+    std::optional<std::vector<Range>> RenameRanges =
+        adjustRenameRanges(Tokens, OldName.getNamePieces().front(),
+                           std::move(FileAndOccurrences.second));
+    if (!RenameRanges) {
+      // Our heuristics fails to adjust rename ranges to the current state of
+      // the file, it is most likely the index is stale, so we give up the
+      // entire rename.
```

Comment:
Why do we give up every file just because one failed? We don't if the buffer doesn't exist, is that case really that different? (I realize this was existing code)

Reply:
Good catch. Changed it to log + continue.